### PR TITLE
Add rules and test for draft 3 to 4 schema keyword

### PR DESCRIPTION
**Pull request by Era-cell**

### CODE REVIEW

File: test/from-draft3/to-draft4/schema.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                "foo": {
+                    "$schema": "https://json-schema.org/draft-03/schema"
+                }
+            }
+        },
+        "to": {
+            "$schema": "https://json-schema.org/draft-04/schema",
+            "$defs": {
+                "foo": {
+                    "$schema": "https://json-schema.org/draft-04/schema"
```
Comment (jviotti) [inline test/from-draft3/to-draft4/schema.json:34]: I don't think you should be transforming this one, as `$defs` is not defined in Draft 3

**Comment by Era-cell** [comment]
@jviotti  I have changed it to properties instead of $defs...

File: test/from-draft7/to-2019-09/schema.json [hunk 2]
```diff
@@ -21,15 +21,15 @@
...
+            "definitions": {
                 "foo": {
                     "$schema": "https://json-schema.org/draft-07/schema"
                 }
             }
         },
         "to": {
             "$schema": "https://json-schema.org/draft/2019-09/schema",
-            "$defs": {
+            "definitions": {
```
Comment (jviotti) [inline test/from-draft7/to-2019-09/schema.json:32]: I think the tests in this file are invalid. On 2019-09 and later, you want to transform `definitions` to `$defs`.

File: test/from-draft7/to-2019-09/schema.json [hunk 3]
```diff
@@ -40,15 +40,15 @@
         "title": "Nested half matching schema",
         "from": {
             "$schema": "https://json-schema.org/draft/2020-12/schema",
-            "$defs": {
```
Comment (jviotti) [inline test/from-draft7/to-2019-09/schema.json:43]: Why are you replacing them with `definitions`? This is a 2020-12 schema, so you should use `$defs`

File: test/from-draft7/to-2019-09/schema.json (hunk 2 above)
Comment (Era-cell) [reply test/from-draft7/to-2019-09/schema.json]: I have removed this file as it wans't supposed to be included in this PR

File: test/from-draft7/to-2019-09/schema.json
Comment (jviotti) [inline test/from-draft7/to-2019-09/schema.json:1]: Was this file accidentally deleted?

Comment (Era-cell) [reply test/from-draft7/to-2019-09/schema.json:1]: No, actually this was draft 3 to 4 related PR, this file got added by mistake... So I deleted it

Comment (Era-cell) [reply test/from-draft7/to-2019-09/schema.json:1]: I had already included tests for this case...

Comment (jviotti) [reply test/from-draft7/to-2019-09/schema.json:1]: Ah cool then

**Review by jviotti — APPROVED** [review]
Looks good but there is a git conflict. Can you resolve?

**Comment by Era-cell** [comment]
@jviotti I am not allowed to resolve conflicts for this PR.. can you help...

**Comment by jviotti** [comment]
@Era-cell What do you mean by "not allowed"? You have permissions to push on this repo (like you have been doing all along)! The typical workflow I do, in case it helps, is:

```sh
# First make sure the target of your PR, `main`, is up-to-date
$ git checkout main
$ git pull

# Go to the conflicted branch and initiate a rebase
$ git checkout draft3To4schema
$ git rebase main

# At this point, git should complain about conflicts and tell you which files are in conflict.
# You can open the conflicted files in your editor and manually fix the conflict
$ code test/from-draft7/to-2019-09/schema.json

# Once it looks good, proceed with the rebase
$ git add .
$ git rebase --continue

# Done! Just force re-push to push your resolution
$ git push -f
```

**Comment by Era-cell** [comment]
@jviotti thank you, now I am able to rebase, previously I resolve conflicts button and squash and merge buttons were disabled.
Right now its good 👍

**Pull request merged**